elytron/src/main/java/org/wildfly/extension/elytron/AdvancedModifiableKeyStoreDecorator.java

```diff
@@ -680,6 +721,245 @@ protected void executeRuntimeStep(final OperationContext context, final ModelNod
         }
     }
 
+    static class ObtainCertificateHandler extends ElytronRuntimeOnlyHandler {
+
+        static final StringListAttributeDefinition DOMAIN_NAMES = new StringListAttributeDefinition.Builder(ElytronDescriptionConstants.DOMAIN_NAMES)
+                .build();
+
+        static final SimpleAttributeDefinition CERTIFICATE_AUTHORITY_ACCOUNT = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.CERTIFICATE_AUTHORITY_ACCOUNT, ModelType.STRING, false)
+                .setMinSize(1)
+                .setRestartAllServices()
+                .setCapabilityReference(CERTIFICATE_AUTHORITY_ACCOUNT_CAPABILITY, KEY_STORE_CAPABILITY, true)
+                .build();
+
+        static final SimpleAttributeDefinition AGREE_TO_TERMS_OF_SERVICE = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.AGREE_TO_TERMS_OF_SERVICE, ModelType.BOOLEAN, true)
+                .setAllowExpression(true)
+                .build();
+
+        static final SimpleAttributeDefinition STAGING = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.STAGING, ModelType.BOOLEAN, true)
+                .setAllowExpression(true)
+                .setDefaultValue(new ModelNode(false))
+                .build();
+
+        static final SimpleAttributeDefinition ALGORITHM = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.ALGORITHM, ModelType.STRING, true)
+                .setAllowExpression(true)
+                .setDefaultValue(new ModelNode("RSA"))
```

Comment:
are the possible values open-ended? Or should we have a validator based on some enum?

Reply:
Only RSA and EC are currently allowed by Let's Encrypt. I've added a validator for this.